src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -571,10 +616,21 @@ internal static bool IsBetterDeletionMatch(FilterResult result1, FilterResult re
             => result1.CompareTo(result2) > 0;
 
         internal static bool MatchesFilterText(
-            CompletionHelper helper, RoslynCompletionItem item,
-            string filterText, CompletionTriggerKind initialTriggerKind,
-            CompletionFilterReason filterReason, ImmutableArray<string> recentItems)
+            CompletionHelper helper,
+            RoslynCompletionItem item,
+            string filterText,
+            CompletionTriggerKind initialTriggerKind,
+            CompletionFilterReason filterReason,
+            ImmutableArray<string> recentItems,
+            bool includeMatchSpans,
+            out PatternMatch? patternMatch)
         {
+            // Get the match of the given completion item for the pattern provided so far. 
+            // A completion item is checked against the pattern by see if it's 
+            // CompletionItem.FilterText matches the item. That way, the pattern it checked 
+            // against terms like "IList" and not IList<>
+            patternMatch = helper.GetMatch(item.FilterText, filterText, includeMatchSpans: includeMatchSpans, CultureInfo.CurrentCulture);
```

Comment:
why do we need the match spans?

Reply:
So we don't need to call this again later when trying to decide the match span for VS items' display text. In majority of the cases, the Roslyn filter text is identical to display text, so we can simply calculate the hightlighted span for VS items based on the match span calculated based on Roslyn item's filter text here.